src/number/numericString.ts

```diff
@@ -0,0 +1,36 @@
+export class NumericString {
```

Comment:
I think you could actually write this whole thing very concisely (less than 10 lines) with 3 separate regexes. Just do grouped matches to grab the values and then send them through parseInt. Something like this:

```{ 
    8: /^0([0-7]+),
    10: /^(\d+)/,
    16: /0x([\da-fA-F]+)/
}```